### PR TITLE
Improve determinism of RNAWithUMIs pipeline

### PR DESCRIPTION
**Pull request by kachulis**
Umi_tools has a bug which leads to ordering of elements in a set being used as a tiebreaker in rare cases.  Since sets are unordered in python, this leads to non-deterministic behavior, which causes our scientific tests to fail.  This pr turns off hash salting in the GroupByUMIs task.  While the python language spec makes no guarantees about the order of elements in a set in any situation, the implementation details of cpython mean that turning off hash salting seems to lead to deterministic behavior (although again, this is not guaranteed, but seems to work in practice).

### CODE REVIEW

**Comment by kachulis** [comment]
Now also updating rnaseqc image to further reduce non-determinism.  Hopefully tests can pass without caching now.

**Review by jessicaway — APPROVED** [review]

**Comment by jessicaway** [comment]
add to whitelist

**Comment by jessicaway** [comment]
test this please

**Comment by jessicaway** [comment]
test this please

**Comment by choover-broad** [comment]
test this please

**Comment by jessicaway** [comment]
add to whitelist

**Comment by kachulis** [comment]
Plumbing and scientific tests now pass without call caching.  I will updated master truth once this is approved and ready to merge so that smart tests pass.

File: tasks/broad/RNAWithUMIsTasks.wdl [hunk 1]
```diff
@@ -622,6 +626,14 @@ task GroupByUMIs {
   command <<<
     bash ~{monitoring_script} > monitoring.log &
 
+    # umi_tools has a bug which lead to using the order of elements in a set to determine tie-breakers in
```
Comment (jessicaway) [inline tasks/broad/RNAWithUMIsTasks.wdl:629]: This is a helpful comment! 👍

**Review by jessicaway — APPROVED** [review]
This is great, thanks so much @kachulis!

File: tasks/broad/RNAWithUMIsTasks.wdl [hunk 2]
```diff
@@ -308,7 +312,7 @@ task rnaseqc2 {
     String sample_id
     File exon_bed
 
-    String docker =  "us.gcr.io/broad-dsde-methods/ckachulis/rnaseqc:2.4.2"
+    String docker = "gcr.io/broad-cga-aarong-gtex/rnaseqc@sha256:627feb33609357a81b5d8aadfed562d60d1292fe364aaec8c86f4d39e1e11417"
```
[the remark this answers is not in the document]
Comment (kachulis) [reply tasks/broad/RNAWithUMIsTasks.wdl:315]: Perhaps.  I think that docker is auto-generated on commits to rnaseqc.

**Review by nikellepetrillo — APPROVED** [review]
this looks good to me, thanks @kachulis!

**Comment by kachulis** [comment]
test this please

**Pull request merged**